qcodes/instrument_drivers/Keysight/keysight_e4980a.py

```diff
@@ -433,6 +435,31 @@ def _get_current_level(self) -> float:
 
         return float(i_level)
 
+    def _is_signal_mode_voltage_on_driver_initialization(self) -> bool:
+        """
+        Checks if signal is set with voltage_level param at instrument driver
+        initialization
+        """
+        assert self.signal_mode() is None
+        try:
+            self.voltage_level()
+            return True
+        except VisaIOError:
+            return False
```

Comment:
In this case I guess we must be in current mode. So perhaps we should ensure that the mode is set correctly too. That could be done by either calling `self.voltage_level` or simply setting the value of signal_mode to `current`